### DISCUSSION TITLE
Is it possible to listen for when getInitialProps is called

### DISCUSSION THREAD

**Discussion by csenio** [vercel/next.js#10809] [Help] [answered]
I use [NProgress](https://github.com/rstacruz/nprogress) (a fake progress bar on top of the page) for user feedback while they wait for page transitions. 

I use it like this right now
```
Router.events.on('routeChangeStart', NProgress.start)
Router.events.on('routeChangeComplete', NProgress.done)
```

Thing is I only need it when I go to a dynamic page since navigating to statically optimized pages is instant anyways. 

Is there a way I can listen to either the start/end of `getInitialProps` or otherwise recognize if the page I'm navigating to is a dynamic page?

**Comment by timneutkens** [comment, marked as answer]
The better solution would be to add a delay to showing it and adding a debounce this is what we do for zeit.co:

```js
import debounce from 'lodash.debounce'
const start = debounce(NProgress.start, 500)
Router.events.on('routeChangeStart', start)
Router.events.on('routeChangeComplete', url => {
  start.cancel()
  NProgress.done()
  metrics.pageview(url)
  trackRenderPerformance()
})
Router.events.on('routeChangeError', () => {
  start.cancel()
  NProgress.done()
})
```
